kinto_emailer/__init__.py

```diff
@@ -48,19 +48,27 @@ def send_notification(event):
             mailer.send_to_queue(message)
 
 
-def _get_collection_record(storage, bucket_id, collection_id):
-    parent_id = '/buckets/%s' % bucket_id
-    record_type = 'collection'
-
-    return storage.get(
-        parent_id=parent_id,
-        collection_id=record_type,
-        object_id=collection_id)
-
-
-def _expand_recipients(storage, recipients):
+def _get_emailer_hooks(storage, context):
+    bucket_id = context['bucket_id']
+    collection_id = context['collection_id']
+    bucket_uri = '/buckets/%s' % bucket_id
+    # Look-up collection metadata.
+    metadata = storage.get(parent_id=bucket_uri,
+                           collection_id='collection',
+                           object_id=collection_id)
+    if 'kinto-emailer' not in metadata:
+        # Try in bucket metadata.
+        metadata = storage.get(parent_id='',
+                               collection_id='bucket',
+                               object_id=bucket_id)
+    # Returns empty list of hooks.
+    return metadata.get('kinto-emailer', {}).get('hooks', [])
+
+
+def _expand_recipients(storage, recipients, context):
     emails = [r for r in recipients if not GROUP_REGEXP.match(r)]
-    groups = [r for r in recipients if GROUP_REGEXP.match(r)]
+    groups = [r.format(**context) for r in recipients
```

Comment:
I think it worth having a comment saying that you might want to build the group name from the `collection_id`